femwell/maxwell/waveguide.py

```diff
@@ -332,7 +381,30 @@ def bform(e, v, w):
     return basis_energy, scipy.sparse.linalg.spsolve(b_operator, a_operator)
 
 
-def calculate_overlap(basis_i, E_i, H_i, basis_j, E_j, H_j):
+def calculate_overlap(
+    basis_i: Basis,
+    E_i: np.ndarray,
+    H_i: np.ndarray,
+    basis_j: Basis,
+    E_j: np.ndarray,
+    H_j: np.ndarray,
+) -> np.complex64:
```

Comment:
I'm just a bit wondering, isn't it always a real number as we calculate it in a symmetric way? :thinking:

Reply:
Not necessarily, as we are operating on complex fields; we extract a real number when we calculate its absolute value squared, which is a measure of power coupling.
As you also reported in https://helgegehring.github.io/femwell/math/maxwell.html, the symmetry is expressed by $c_{\mu \nu}=c_{\nu \mu}^*$. This is a consequence of Lorentz's reciprocity theorem, which states that the scattering matrix of a linear non-magnetic device equals its adjoint: $S=S^{\dagger}$.
I recommend https://doi.org/10.1038/nphoton.2013.185, as it explains the concept very well!

Reply:
Amazing, thanks, Ill have a read! :)